pulpcore/cli/common/generic.py

```diff
@@ -435,3 +456,155 @@ def label_show(entity_ctx: PulpEntityContext, key: str) -> None:
         label_group.add_command(subcmd)
 
     return label_group
+
+
+def repository_content_command(**kwargs: Any) -> click.Group:
+    """A factory that creates a repository content command group."""
+
+    if "name" not in kwargs:
+        kwargs["name"] = "content"
+    content_contexts = kwargs.pop("contexts", [])
+    content_names = [content.ENTITY.split()[0] for content in content_contexts]
```

Comment:
I think, this is a tiny little bit too much magic for my liking.
This `ENTITY` field is only used to report failures to the user by now. I'm not even sure whether it should be translated eventually.

Turns out, i started to need a similar thing in the other PR:
https://github.com/pulp/pulp-cli/pull/161/files#diff-2330f1d2bec4c88428af0e90f4ad6feff6a1663af7be7b9fedb645ed98846f68R137
https://github.com/pulp/pulp-cli/pull/161/files#diff-4833716c6e1661345a3aebb74ab1c3446ab85979e75bdd93e3f538cbb58cfb14R115

Would that apply here too?